contracts/netmap/contract.go

```diff
@@ -278,6 +302,37 @@ func AddPeer(nodeInfo []byte) {
 	}
 }
 
+// AddNode adds a new node into the candidate list for the next epoch. Node
+// must have Online state to be considered and the request must be signed by
+// both the node and Alphabet. AddNode event is emitted upon success.
+func AddNode(n Node2) {
+	ctx := storage.GetContext()
+
+	if n.State != nodestate.Online {
+		panic("can't add non-online node")
+	}
+
+	common.CheckWitness(n.Key)
+	common.CheckAlphabetWitness(common.AlphabetAddress())
```

Comment:
That's intentional, the old code was just wrong and the whole addPeer/addPeerIR story has ended quite some time ago. The model we have now is much simpler and it will work fine.